src/main/java/frc/robot/subsystems/TrapArmSubsystem.java

```diff
@@ -0,0 +1,224 @@
+    scoringMotor.getPIDController().setFF(scoringPID[4]);
+
+    scoringBreak = new DigitalInput(TrapArmConstants.scoringBreak_ID);
+
+    // SmartDashboard
+    SmartDashboard.putBoolean("Arm Zeroed", (baseBreak.get() && scoringBreak.get()));
+  }
+
+  // Commands
+  public Command intakeSource() {
+    return startEnd(
+        () -> {
+          if (!sourceBreak.get()) {
+            setTrapArm(TrapArmState.FROM_SOURCE);
+            rollerMotor.set(TrapArmConstants.rollerIntakeSpeed);
+          }
+        },
+        () -> {
+          rollerMotor.stopMotor();
+        });
+  }
+
+  public Command intakeGround() {
+    return startEnd(
+        () -> {
+          if (!groundBreak.get()) {
+            setTrapArm(TrapArmState.FROM_INTAKE);
+            rollerMotor.set(-TrapArmConstants.rollerIntakeSpeed);
+          }
+        },
+        () -> {
+          rollerMotor.stopMotor();
+        });
+  }
+
+  public Command scoreAMP() {
+    return startEnd(
+        () -> {
+          setTrapArm(TrapArmState.AMP_SCORE);
+          rollerMotor.set(-TrapArmConstants.rollerScoringSpeed);
+        },
+        () -> {
+          rollerMotor.stopMotor();
+        });
+  }
+
+  public Command scoreTrap() {
+    return startEnd(
+        () -> {
+          setTrapArm(TrapArmState.TRAP_SCORE);
+          rollerMotor.set(TrapArmConstants.rollerScoringSpeed);
+        },
+        () -> {
+          rollerMotor.stopMotor();
+        });
+  }
+
+  public Command zeroArm() {
+    return startEnd(
+      () -> {
+        if (!baseBreak.get()) {
+          baseMotor1.getPIDController().setReference(10, ControlType.kVelocity);
+          baseMotor2.getPIDController().setReference(10, ControlType.kVelocity);
+        } else {
+          baseMotor1.stopMotor();
+          baseMotor2.stopMotor();
+        }
+        if (!scoringBreak.get()) {
+          scoringMotor.getPIDController().setReference(10, ControlType.kVelocity);
+        } else {
+          scoringMotor.stopMotor();
+        }
+      },
+
+      () -> {
+        baseMotor1.stopMotor();
+        baseMotor2.stopMotor();
+        scoringMotor.stopMotor();
+      }
+    );
+  }
+
+  public Command setTrapArm(TrapArmState state) {
+    return startEnd(
+        () -> {
+          wristMotor.getPIDController().setReference(state.getWristPose(), ControlType.kPosition);
+          baseMotor1.getPIDController().setReference(state.getBasePose(), ControlType.kPosition);
+          baseMotor2.getPIDController().setReference(state.getBasePose(), ControlType.kPosition);
+          scoringMotor
+              .getPIDController()
+              .setReference(state.getScoringPose(), ControlType.kPosition);
+        },
+        () -> {
+          wristMotor.stopMotor();
+          baseMotor1.stopMotor();
+          baseMotor2.stopMotor();
+          scoringMotor.stopMotor();
+        });
+  }
+
+  @Override
+  public void periodic() {
+    SmartDashboard.putBoolean("Source Beam Break", sourceBreak.get());
```

Comment:
Do we want these to be DebugEntry instead?

Reply:
That might make more sense

Reply:
done